test/asset/asset-list-loader.test.mjs

```diff
@@ -0,0 +1,255 @@
+import { Application } from '../../src/framework/app-base.js';
+import { AssetListLoader } from '../../src/asset/asset-list-loader.js';
+import { Asset } from '../../src/asset/asset.js';
+
+import { HTMLCanvasElement } from '@playcanvas/canvas-mock';
+
+import { expect } from 'chai';
+
+describe('AssetListLoader', function () {
+
+    let app;
+    const assetPath = 'http://localhost:3000/test/test-assets/';
+
+    beforeEach(function () {
+        const canvas = new HTMLCanvasElement(500, 500);
+        app = new Application(canvas);
+    });
+
+    afterEach(function () {
+        app.destroy();
+    });
+
+    describe('#constructor', function () {
+        it('instantiates correctly', function () {
+            const assetListLoader = new AssetListLoader([], app.assets);
+            expect(assetListLoader).to.be.ok;
+        });
+        it('stores a single asset', function () {
+            const assets = [
+                new Asset('model', 'container', { url: `${assetPath}test.glb` })
+            ];
+            const assetListLoader = new AssetListLoader(Object.values(assets), app.assets);
+            expect(assetListLoader._assets[0].name).to.be.equal('model');
+        });
+        it('stores multiple assets', function () {
+            const assets = [
+                new Asset('model', 'container', { url: `${assetPath}test.glb` }),
+                new Asset('styling', 'css', { url: `${assetPath}test.css` })
+            ];
+            const assetListLoader = new AssetListLoader(assets, app.assets);
+            expect(assetListLoader._assets[0].name).to.be.equal('model');
+            expect(assetListLoader._assets[1].name).to.be.equal('styling');
+        });
+        it('stores single copies of duplicated assets', function () {
+            const assets = [
+                new Asset('model', 'container', { url: `${assetPath}test.glb` })
+            ];
+            const assetListLoader = new AssetListLoader([assets[0], assets[0]], app.assets);
+            expect(assetListLoader._assets.length).to.be.equal(1);
+        });
+        it('adds the supplied registry to any assets that do not have one', function () {
+            const assets = [
+                new Asset('model', 'container', { url: `${assetPath}test.glb` })
+            ];
+            expect(assets[0].registry).to.be.equal(null);
+            const assetListLoader = new AssetListLoader([assets[0], assets[0]], app.assets);
+            expect(assetListLoader._assets[0].registry).to.be.equal(app.assets);
+        });
+    });
```

Comment:
Minor point but the whitespace formatting around `describe`/`it` blocks doesn't match the other tests. Please try to match the formatting elsewhere.

Reply:
Done 👍